app/lib/utils/decoder_factory.rb

```diff
@@ -0,0 +1,22 @@
+require_relative "secret_decoder"
+require_relative "jwks_decoder"
+class DecoderFactory
+  cattr_reader :decoders
+  class << self
+    # Any new decoders should be added here:
+    @@decoders = [ JwksDecoder.new(Config[:jwks_url]),
+                   SecretDecoder.new(Config[:jwt_signing_key]) ]
+
+    def get(config)
+      validateConfig(config)
+      decoder = decoders.find { |c| c.configured?(config) }
+    end
+
+    private
+    def validateConfig(config)
+      if decoders.filter  { |c| c.configured?(config) } .length != 1
```

Comment:
it seems like this loop could also return the one decoder found, so you don't have to loop through a second time? not big deal

Reply:
done